### PR TITLE
Support format in transformation API

### PR DESCRIPTION
**Pull request by tocker**

### CODE REVIEW

File: test/api_spec.js [hunk 1]
```diff
@@ -509,62 +516,83 @@ describe("api", function() {
...
+          ]);
+          expect(transformation.used).not.to.be.ok();
+        });
       });
-    });
-    it("should allow listing of named transformations", function() {
-      return helper.mockPromise(function(xhr, write, request) {
-        cloudinary.v2.api.transformations({
-          named: true
+      it("should allow creating named transformation with format", function() {
```
Comment (mayashavin) [inline test/api_spec.js]: empty format

File: test/utils_spec.js [hunk 2]
```diff
@@ -907,48 +907,83 @@ describe("utils", function () {
...
+      ['should support transformations with multiple components and format',
+        [{transformation: [scaled(), sepia()], format: 'gif'}, sepia()],
+        'c_scale,h_200,w_100/c_lfill,e_sepia,w_400/gif|c_lfill,e_sepia,w_400'],
+    ].forEach(function ([subject, input, expected]) {
+      it(subject, function () {
+        expect(utils.build_eager(input)).to.eql(expected);
+      })
+    });
+    it("build_explicit_api_params should support multiple eager transformations with a pipe", function () {
+      var options = {
```
Comment (mayashavin) [inline test/utils_spec.js]: maybe move outside?

File: lib-es5/api.js [hunk 3]
```diff
@@ -392,40 +392,36 @@ exports.transformations = function transformations(callback) {
 exports.transformation = function transformation(transformation, callback) {
   var options = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
 
-  var uri = void 0;
-  uri = ["transformations", transformationString(transformation)];
-  return call_api("get", uri, only(options, "next_cursor", "max_results"), callback, options);
+  var params = only(options, "next_cursor", "max_results");
+  params.transformation = utils.build_eager(transformation);
+  return call_api("get", ["transformations"], params, callback, options);
```
Comment (mayashavin) [inline lib-es5/api.js]: make it constant variable

File: test/utils_spec.js [hunk 4]
```diff
@@ -907,48 +907,83 @@ describe("utils", function () {
...
+      ['should concatenate format at the end of the transformation',
+        ([scaled({format: 'gif'}), sepia()]),
+        'c_scale,h_200,w_100/gif|c_lfill,e_sepia,w_400'],
+      ['should concatenate format at the end of the transformation',
+        ([scaled({format: 'gif'}), sepia({format: 'jpg'})]),
+        'c_scale,h_200,w_100/gif|c_lfill,e_sepia,w_400/jpg'],
+      ['should support transformations with multiple components and format',
+        [{transformation: [scaled(), sepia()], format: 'gif'}, sepia()],
+        'c_scale,h_200,w_100/c_lfill,e_sepia,w_400/gif|c_lfill,e_sepia,w_400'],
+    ].forEach(function ([subject, input, expected]) {
```
Comment (const-cloudinary) [inline test/utils_spec.js]: Add test for empty format
